R/lint.R

```diff
@@ -182,20 +182,12 @@ lint_dir <- function(path = ".", ...,
     return(lints)
   }
 
-  pb <- if (isTRUE(show_progress)) {
-    txtProgressBar(max = length(files), style = 3L)
-  }
-
   lints <- flatten_lints(lapply(
-    files,
-    function(file) {
-      maybe_report_progress(pb)
-      lint(file, ..., parse_settings = FALSE, exclusions = exclusions)
-    }
+    # NB: This cli API is experimental (https://github.com/r-lib/cli/issues/709)
+    cli::cli_progress_along(files, name = "Running linters"),
```

Comment:
is this no longer conditional on `isTRUE(show_progress)`?

Reply:
Good catch!

- `show_progress = TRUE`

<img width="979" alt="Screenshot 2024-12-02 at 08 49 13" src="https://github.com/user-attachments/assets/9ec6cb1c-6f8d-4328-9870-09807f7445c1">

- `show_progress = FALSE`

<img width="974" alt="Screenshot 2024-12-02 at 08 45 45" src="https://github.com/user-attachments/assets/bfbfa566-fe18-46a0-8ac3-76f04dec8f99">